index.d.ts

```diff
@@ -241,8 +241,14 @@ export class Schema {
   path(path: string, constructor: any): Schema;
   pathType(path: string): string;
   plugin(plugin: (schema: Schema, options?: Object) => void, options?: Object): Schema;
-  post(method: string, fn: Function): Schema;
-  pre(method: string, callback: Function): Schema;
+  pre(method: string, fn: HookSyncCallback): Schema;
+  pre(method: string, fn: HookSyncCallback, errorCb: HookErrorCallback): Schema;
```

Comment:
Can't these two overloads be one function with the last argument being optional?